lib/node_modules/@stdlib/blas/base/cscal/README.md

```diff
@@ -0,0 +1,243 @@
+var imagf = require( '@stdlib/complex/imagf' );
+
+var x = new Complex64Array( [ 1.0, 1.0, 1.0, 1.0, 1.0, 1.0 ] );
+var cx = new Complex64Array( [ 2.0, 0.0 ] );
+cscal( 3, cx, x, 1 );
+
+var z = x.get( 0 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 2.0
+
+var im = imagf( z );
+// returns 2.0
+```
+
+The function has the following parameters:
+
+-   **N**: number of indexed elements.
+-   **cx**: complex constant 
+-   **x**: input [`Complex64Array`][@stdlib/array/complex64].
+-   **strideX**: index increment for `x`.
+
+The `N` and stride parameters determine how values from `x` are scaled by `cx`. For example, to scale every other value in `x` by `cx`,
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+var cx = new Complex64Array( [ 2.0, 0.0 ] );
+
+cscal( 2, cx, x, 2 );
+
+var z = x.get( 2 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns 10.0
+
+var im = imagf( z );
+// returns 12.0
+```
+
+Note that indexing is relative to the first index. To introduce an offset, use [`typed array`][mdn-typed-array] views.
+
+<!-- eslint-disable stdlib/capitalized-comments -->
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+// Initial arrays...
+var x0 = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0 ] );
+var cx = new Complex64Array( [ 2.0, 2.0 ] );
+
+// Create offset views...
+var x1 = new Complex64Array( x0.buffer, x0.BYTES_PER_ELEMENT*1 ); // start at 2nd element
+
+// Copy in reverse order every other value from `x1` into `y1`...
+cscal( 3, cx, x1, 1 );
+
+var z = x0.get( 1 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns -2.0
+
+var im = imagf( z );
+// returns 14.0
+```
+
+#### cscal.ndarray( N, cx, x, strideX, offsetX )
+
+Scales values from `x` by `cx` using alternative indexing semantics.
+
+```javascript
+var Complex64Array = require( '@stdlib/array/complex64' );
+var realf = require( '@stdlib/complex/realf' );
+var imagf = require( '@stdlib/complex/imagf' );
+
+var x = new Complex64Array( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
+var cx = new Complex64Array( [ 2.0, 2.0 ] );
+
+cscal.ndarray( x.length, cx, x, 1, 0 );
+
+var z = x.get( 0 );
+// returns <Complex64>
+
+var re = realf( z );
+// returns -2.0
+
+var im = imagf( z );
+// returns 6.0
+```
+
+The function has the following additional parameters:
+
+-   **offsetX**: starting index for `x`.
+
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on starting indices. For example, to scale every other value in `x` starting from the second value by `cx` where `x[i] = cx[0] * x[i]`, `x[i+1] = cx[1] * x[i+1]`,...,
```

Comment:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on starting indices. For example, to scale every other value in `x` starting from the second value by `cx` where `x[i] = ca * cx[i]`, `cx[i+1] = ca * cx[i+1]`,...,
```

Use BLAS naming conventions here and elsewhere. https://github.com/OpenMathLib/OpenBLAS/blob/develop/lapack-netlib/BLAS/SRC/cscal.f

Reply:
Also, there seem to be a number of copy-paste errors in this PR.